datadog/dogstatsd/base.py

```diff
@@ -274,24 +297,66 @@ def __init__(
         self._reset_telemetry()
         self._telemetry_flush_interval = telemetry_min_flush_interval
         self._telemetry = not disable_telemetry
+        self._last_flush_time = time.time()
 
+        self._current_buffer_total_size = 0
+        self._buffer = []  # type: List[Text]
         self._buffer_lock = RLock()
 
+        self._reset_buffer()
+
+        # This lock is used for backwards compatibility to prevent concurrent
+        # changes to the buffer when the user is managing the buffer themselves
+        # via deprecated `open_buffer()` and `close_buffer()` functions.
+        self._manual_buffer_lock = RLock()
+
+        # If buffering is disabled, we bypass the buffer function.
+        self._send = self._send_to_buffer
+        if disable_buffering is True:
+            log.info("Statsd buffering is disabled")
+            self._send = self._send_to_server
+
+        # Start the flush thread if buffering is enabled and the interval is above
+        # a reasonable range. This both prevents thrashing and allow us to use "0.0"
+        # as a value for disabling the automatic flush timer as well.
+        if disable_buffering is False and flush_interval >= MIN_FLUSH_INTERVAL:
+            self._register_flush_thread(flush_interval)
+            log.debug(
+                "Statsd flush thread registered with period of %s",
+                flush_interval,
+            )
+        else:
+            log.info("Statsd periodic buffer flush is disabled")
```

Comment:
Don't we get in that else if `disable_buffering is set` ? I think it would make more sense if it was only printed when `disable_buffering` is false.

Reply:
The message gets shown for either:
- Buffering being explicitly disabled
- Or the flush interval is below the minimum allowed (explicit flush thread disabling)
So I think the message gets shown for the appropriate conditions